src/navigation/actions.js

```diff
@@ -126,35 +143,27 @@ export const createSupplierRequisition = ({
 };
 
 /**
- * Action creator for navigating to a customer invoice. Ensures the CI is at least
- * confirmed before navigating as if this is not enforced, it is possible for
- * a particular item being issued across multiple invoices in larger quantities
- * than are available.
+ * Creates a stocktake and replaces the current route in the StackNavigator which
+ * would be a StocktakeManagePage, with a StocktakeEditPage, with the newly
+ * created Stocktake.
  *
- * @param {Object} transaction The CI to navigate to.
- * @param {Object} dispatch    Redux dispatch method.
+ * @param {Object} StocktakeParameters  Parameters for the stocktake to create.
  */
-export const gotoCustomerInvoice = transaction => dispatch => {
-  const { isConfirmed, isFinalised } = transaction;
+export const createStocktake = ({ currentUser, stocktakeName, program, itemIds }) => dispatch => {
+  let stocktake;
 
-  // Customer invoices are generally created with the status confirmed. This handles unexpected
-  // cases of an incoming sycned invoice with status 'nw' or 'sg'.
-  if (!isConfirmed && !isFinalised) {
-    UIDatabase.write(() => {
-      transaction.confirm(UIDatabase);
-      UIDatabase.save('Transaction', transaction);
-    });
-  }
+  UIDatabase.write(() => {
+    stocktake = createRecord(UIDatabase, 'Stocktake', currentUser, stocktakeName, program);
+    if (program) stocktake.addItemsFromProgram(UIDatabase);
+    else if (itemIds) stocktake.setItemsByID(UIDatabase, itemIds);
```

Comment:
the items from program is 'add' and the other is 'set'
but within `setItemsByID` it talks about `itemsToAdd` and creates records. I'd suggest changing that one to `addItemsById`